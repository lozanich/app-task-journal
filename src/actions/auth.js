import Swal from "sweetalert2";
import { firebase, googleAuthProvider } from "../firebase/firebaseConfig";
import { types } from "../types/types";
import { endLoadingAction, startLoadingAction } from "./ui";

export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {
    // setTimeout(() => {
    //   dispatch(login(987456, "Lozano"));
    // }, 3500);
    dispatch(startLoadingAction());
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        console.log(user);

        const { uid, displayName } = user;
        dispatch(login(uid, displayName));
        dispatch(endLoadingAction());
      })
      .catch((e) => {
        console.log(e);
        dispatch(endLoadingAction());
        Swal.fire("Error", e.message, "error");
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    console.log(email, password, name);
    console.log(dispatch);
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });

        const { uid, displayName } = user;
        dispatch(login(uid, displayName));
      })
      .catch((e) => {
        console.log(e);
        Swal.fire("Error", e.message, "error");
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        const { uid, displayName } = user;
        dispatch(login(uid, displayName));
      });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};

export const startLogout = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signOut()
      .then(() => {
        dispatch(logout());
      });
  };
};

export const logout = () => {
  return {
    type: types.logout,
  };
};
