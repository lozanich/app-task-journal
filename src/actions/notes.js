import Swal from "sweetalert2";
import { db } from "../firebase/firebaseConfig";
import { fileUpload } from "../helpers/fileUpload";
import { loadNotes } from "../helpers/loadNotes";
import { types } from "../types/types";

// react_journalapp

export const startNewNote = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;

    const newNote = {
      title: "",
      body: "",
      date: new Date().getTime(),
    };

    const docRef = await db.collection(`${uid}/journal/notes`).add(newNote);
    dispatch(setActiveNote(docRef.id, newNote));
    newNote.id = docRef.id;
    dispatch(addNote(newNote));
  };
};

export const setActiveNote = (id, note) => ({
  type: types.notesActive,
  payload: {
    id,
    ...note,
  },
});

export const addNote = (note) => ({
  type: types.notesAddNew,
  payload: {
    note,
  },
});

export const setNotesLoaded = (uid) => {
  return async (dispatch) => {
    const notes = await loadNotes(uid);
    dispatch(loadNotesAction(notes));
  };
};

export const loadNotesAction = (notes) => ({
  type: types.notesLoad,
  payload: { notes },
});

export const startSaveNote = (note) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;

    if (!note.url) {
      delete note.url;
    }

    const noteToFirestore = { ...note };
    delete noteToFirestore.id;
    try {
      await db.doc(`${uid}/journal/notes/${note.id}`).update(noteToFirestore);
      Swal.fire("Saved", note.title, "success");
      dispatch(refreshNote(note.id, note));
    } catch (error) {
      Swal.fire("Error", error.message, "error");
    }
  };
};

export const refreshNote = (id, note) => {
  return {
    type: types.notesUpdated,
    payload: {
      id,
      ...note,
    },
  };
};

export const startUploading = (file) => {
  return async (dispatch, getState) => {
    const { active: aciveNote } = getState().notes;

    Swal.fire({
      title: "Uploading...",
      text: "Please wait...",
      allowOutsideClick: false,
      onBeforeOpen: () => {
        Swal.showLoading();
      },
    });

    const url = await fileUpload(file);
    aciveNote.url = url;
    dispatch(startSaveNote(aciveNote));

    Swal.close();
  };
};

export const startDelete = (id) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    const url = `${uid}/journal/notes/${id}`;

    try {
      db.doc(url).delete();
      dispatch(deleteNote(id));
      Swal.fire({
        title: "Eliminada",
        text: "Nota eliminada correctamente",
        icon: "success",
      });
    } catch (error) {
      console.log(error);
    }
  };
};

export const deleteNote = (id) => ({
  type: types.notesDelete,
  payload: { id },
});

export const notesLogoutClean = () => ({
  type: types.notesLogoutCleanning,
});
