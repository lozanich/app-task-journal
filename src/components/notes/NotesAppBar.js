import React from "react";
import moment from "moment";
import { useDispatch, useSelector } from "react-redux";
import { startSaveNote, startUploading } from "../../actions/notes";

export const NotesAppBar = () => {
  const dispatch = useDispatch();
  const { active: note } = useSelector((state) => state.notes);
  const dateLocal = moment(note.date);
  // console.log(note)

  const handleSaveNote = () => {
    console.log("savvinf note");
    dispatch(startSaveNote(note));
  };

  const handleUploadImage = () => {
    document.querySelector("#fileSelector").click();
  }

  const handleFileChange = (e) => {
    const file = e.target.files[0];
    if (file) {
      console.log('Realiza la accion')
      dispatch(startUploading(file))
    }
  }

  return (
    <div className="notes__appbar">
      <span>{dateLocal.format("dddd, MMMM Do YYYY, h:mm:ss a")}</span>

      <input
        id="fileSelector"
        type="file"
        name="file"
        style={{ display: "none" }}
        onChange={handleFileChange}
      />

      <div>
        <button
          onClick={handleUploadImage}
          className="btn">Picture</button>
        <button onClick={handleSaveNote} className="btn">
          Save
        </button>
      </div>
    </div>
  );
};
