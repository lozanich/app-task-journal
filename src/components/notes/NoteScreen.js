import React from "react";
import { NotesAppBar } from "./NotesAppBar";

export const NoteScreen = () => {
  return (
    <div className="notes__main-content">
      <NotesAppBar />

      <div className="notes__content">
        <input
          className="notes__title-input"
          type="text"
          placeholder="Title"
          autoComplete="off"
        ></input>

        <textarea
          placeholder="What happend today?"
          className="notes__textarea"
        ></textarea>

        <div className="notes__image">
          <img
            alt="images"
            src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAoHCBYWFRgVFRYYGBgaGBgYGhoaGhwaHBgaGhoaGhgYGhocIS4lHB4rIRgYJjgmKy8xNTU1GiQ7QDs0Py40NTEBDAwMEA8QGhISHzQkISE0NDQ0NDQxNDQ0NDQ0NDQ0NDQ0NDQ0NDE0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NDQ0NP/AABEIALcBEwMBIgACEQEDEQH/xAAbAAABBQEBAAAAAAAAAAAAAAAEAAECAwUGB//EAEgQAAIBAgMDCQQHBQYEBwAAAAECAAMRBBIhBTFBBhMiUWFxgZGhMkKxwRRScpLR4fAjYoKishUWM1PS8QckwvJDVGODk9Pi/8QAGQEBAQEBAQEAAAAAAAAAAAAAAAECAwQF/8QAJxEBAAICAgEDBAIDAAAAAAAAAAERAhIDITFBUWEEEzKRM7EUcYH/2gAMAwEAAhEDEQA/AN0VzJLiIBmks08+j0WONUjcSPGSOMdQLDUDRrkHv0meK0s5yTUtu4bb1VSFJD6e93a2I1HjffNjBbfRjZxkJ3XN1P8AFw8bTiSbxLpum4mYZnGJeniKcBh9q1EFlqMo0sLBgOyzDQTWwvKi2lQBh9ZQQfFTofMTpGUOc4y6mK8yF5Q4c++fuP8A6ZQ/KeiDort2hQPiwMbQmst2KYg5T0OJYd6/nB6/K2iB0VZj22Uedz8I2hal0cV5yJ5Z/wDojs/af/iDNyyqcEQfePzEbQay7i8U85rco8QxvzuXsVVAHmCfMwSpjKjG7VXJ7Xb8ZNjV6atVSStxcaEXFxoDqOGhB8Y5nly4moHVw7Z0vlYm5ANri53g2Gh6p0uE5V6nnVsMtxluSWv7PcRx/wB42NXVERiIBhtsUahAR1JIvY3U+TW64XnluCjkxs8iXkS8llLQ8YvKS8iakmzWq4tImVc5FmkspIrGKRs0WaLWkSkbLLA0V5FVlI0sjGFQvFmiIkDIUnmilV4oWnBlogbyq8kzgKTabpzs5aTD62gQfpkdaj5mFKBxiixCaxEGVL1wh3AiltQzxK/bKGaRMtJYg1vGUvWMheNaKLPmjZo0reqBFFrCTIO9t5g9TEeEDqVh1y0DmrjtkPpwG74iZpePYxQ1l2iONxLPpy9ZmJYyQB4RQ2higdxB7JZTxrqQQzaWtZmFrXtYg8MzeZmEbyaYgjjcdv4xUJb0DY/KdSESrfMSwZyRYLYlSdNeA9Z0K4pDqHQj7Q47p5KmIB7ISjW1BicDZ6lVroouzKBrvIG7fM2pt+kM1rtbQWHtG1z2AXsL/KcEtfXqlornrvJobOnxHKRz/hoo7XJPoLfGZ1ba2IcWNTKOpQo9bX9ZliqYjXMuhs0BtGuDcVX3W3g+hEITlBiAd6NpaxXu16NtdPWYhxJi+kRqmzosDykdRlqLn/eGh8QND4TYwG26dQDpBH4o2nkdxnDCvHNQGJwWMno61Li4II4EG4PjHzTiNmbZeiGVbMp1AbcG69NeA0mzhuUyEdNSpsPZ1udb7zpw48TMTjMNRlDdzxZpgVuVFMewjN1XIW/XuvbjATymfOzZFykABCbhSN5uACSflJrk1tDrtI05L+9L/UT+b8YpdZNoBvTFtN8CxqkKOs+U0SsExtPNlvuB18bds1PhzjyBwqBnNjxPboun4TUNPqtaCYZMrff6vrWHw+MIdzEGR3YcJSYrRNpAgxiJlVTEAcRB22ivWJbgqZFSDVQO2CNiw0FxeKsIsjGRrYi+4iUVHtvN5hVMeb75E7RMltULr4wg74K2MECrYi8hRR3YIiM7HcqKWY9yrcmBrUsUvAa9svXE34yhOT2JGtRUojrrVadO3ersH9JfS2bSQ3qY7DDspmpUP8qW9ZYSZhPNeRIhHP4FdDjHb7FBj6s4jf2ls4b6uJbupIvxczTNqcx64hU65Z/auzvr4r7tP8YhtTZp9/FfdpylmMSVGXUS9Mdsw76uKHelP/VLlfZ7ezjKifbo5h/K8HSFLFg79D6S7NfdCqOwaNQXo43Dv9sPTPlraOeTGJXVGpP2JUQ+j5by2gUVT1yS1euRr4erT/xqbp2spCnua1j5yu99RKi0vFeV3klI4wJZpLvklHVHtAjeK8WUnhKMRTa1/Qa3gX3jZjxgj4nQWOnXuB0vLMPVzHh8/wDaKgX85FJ82OyNHS2sG20ldTaqMRu0v56WnJvVlfPTy3lLrERDtsPiwbajXu3j/uMNFiL3nGYLG290HumuNr2XRP14SxlXlJxttuR2TF2hj7XtbzmdidpVDu0mfiKzEazW1pVIYnaJJg64rXWB1t8igJ3S0bNqljbDfBcRiieMFRG6jCadEcZKLCF4xeaIKjgJS9LOQFW5JAAGpJJsAO2Vm0Nl7PfEVVpUxdmPgBxY9gnplfZVLC0eaQmyr+0cMRzj8SwBFwNwB3TR5L8n02fQzOAa7gFzvy8Qg7BvPWROR5WbTzHID2t8bTWMWxll6OaxuHVmJWwF+AH4zKq0ukVFj6Q137YNhtWLGbkPhMEue1dnppuzKufpb1Wy33jN17ppNszBMejjQBYAZ0dT45kA693+4NSp7BP1j5lbfOZuJ9tu8/nESVM+ojGbPZCxS7op0e1sw+tl3gfKP/ZVTiFH8QPwvCNjYo5ijG6kG19bEcO619OyH4eqqpZj7DZLmwutr0z93o/wGSZSdmWNkPxKDz/CAMpBI6iR5aTpaeMQkBWUknrmNtCnlqNcDU5hbdY6yRKxfq6H/h3hecxAUuVU3uQd9hlUffdJ6g/JZW3VDvOoNQag2IutUcdJ5dyZTJSNTUZWptcGxNqnOALcakthkv1Ak9k91wyDKALZeHdbSakcdgNmVXq1qSYmrRaiyreyulRXRHVirgtbpkaN7pmZtrAvQZFxKUwlRgiYqgpRUdvZGIoksArH3lP4TsFphNoDX/Gw27heg/SbvtWUeAmptXZqYik9GqMyVFKsO/cR1EGxB6wJmZWHj1asyOyVFs6kqw6iDa35yFfE6plNukM1+q/w3+kJ2ng3NN8+uJwjLh8QeNSnb/lsRvucyZVJOugnPKON9ZNpbqJh0lM8Ru7N1pcjg8LGYNKoy6jT4dW6E0toW0axPZv8pu2KbJmfimY6aA20vx/VvWRGKU7zac/isQS7ANqrHUnr4dnH0i1iBVDFarwy66nS/E/KalDEqxLC2aw/AfKcojk2HH85p7KrdOzG3S3D3rHdfhFlOpSoLRQek7WGg8LRSox02Or+/b9d0s/u4v8Amr+vCHYXFKRrTU+EuqOhHsKPD8p5rl1D4XBIm90bst+cNqYhLarfwt85lYhVvpp4D8JRd/D9dkusT2WNd6d7lCP13Rc1QbeD+u+CJUYm2UE7rakn8ZtYPklia2ow7KDxfoDv6RBPgJdUtkvs2jvC38AfnM3GOKAHN07te7MUVso91UVgwHstdiL7gLa37ul/w4qj2qtNO5mY+WnxhVPkPzZuK6OGRkcZxTO9WUgkNuIM1HlmZhwbDPZxTVMyqSua1myjPYBbAFrm3C9oBiUce6befynruC5IYUAF2QNxAqM4HjmAP3ZoLsbAL7TIe/J+F4Lh4KHN9x8p6l/w25O5B9LqqC1v2KHUqDvqMBcgncNN1zxE6ymmAT2Dhh9qzehItNDD4/DL7Nal3KUUeFtfWO0uHPbZoYioxyBDYWtd7i+8kBCdflOEfk1Vclg9N7kkkLXKjvYIAPOeyfS6B3PTJ7WU/OVoM+YOyuD7IV7C3aNPnLcs1DxHE8mXTRnog9XTP4iUYnk3UpqHDoUbcwTeeqxNxPR9tcnsTULBKFFVLEgoVD2vpdmOp65lYrk5iGQJURkYLZbWZDbd7N7Hsmk7eevs5yP8Wl3WZSPNLcOuBvs17k3Q9x/KdnU5GYq9ubPfY/AC/pJryCxZFyEUfvOF9DNdLbh12dVBuqajipF5I5aZU1AGqHWzksEUGy3VdWY2J10AtoSej0W2dinDLmerRqPey01e5v8AWa3uDjbU6DS9xdhtgI6BjVz5ukSUbKzfWytkI7NN1raRVlua2kEcI9JWIZTnXKSEcMQchJJykWNjuuRutASXOhuTawve/drPQ8NyfUDLnsBwVALeZMPwfJ+kCCzO3YStvRYpNocbtBslKlSG65fXiMlOmt/4lq+c9m5M7ZpPhaDvVQMaSBruL5lXK2m/epnOf2AjKVFJ7XO4vvJzEgi1rk3sOuE4bYSqLcw57SanDdvMk492txVNTa+0KfPYWqjq4FV6bZSCQr03Y6b/AGqSToMNXV1DIwIPmOwjhOTOxlHTejovSu7MAtve6TWELwWAJGZEUg3sVfQi/Aq3WPSTX5S2bysVKWNw9cWZa3/I4mmNSyVNabFb8Gvrv1A4zzvlDs18JiHotdlHSRj79NvZbv3g9oM9ep7KXOHahTLBgwZgGIYbmBJOo65ftjYlHFJkrpcWJV10ZCbXCn5ag5dZmYbiXhq1GZej3yKowT2DqQcxXXwb9b50vKDki2EPt9Bj0Hymx/dP1W7PKc86vmvlLW0zAkzLaFVjYXvprp8IBl1PUePdC6vPBr5SfX03yOLZ7ByMu7hpfrliZhJV4PD7mv2/lDMDSRXzsdxvbr7oMtcFQCbW6hYExNpe5vcC0kzI2aeKWw0HlHmMtNjwil3ldYbtLDEcPWGLS/d9fyl9LZr31LDsJ1lxwTDj5kD5znNr0Aeh2AfrulLUz+h+U0atB+BHnf4QUU3U/wDd+EsWk0qw22XoPzdFX51iLFWRGYNoFUvSduFzYrob620tx2MxasgfEVEzEqyMRUdSELBsxCBgbEG4FiR7VzaygifSaNRzrZ10FyTkOQa8T0rd0s5QYNOZpvTz52cIiVCFJASoWdszdEdugsRprc98Y6cpntlNjcQdfpJIJy3Kgak2ANmPXKmxGJ1/bobMFN7ixNrbu8ecoGAq6AJcA3OVlY34aAxDCV9xpVNXDGyMbZbW3D90SnSbVMTe2ekTmy6sd9r23aaS6pgsYFzFKWW182c23X7724WvoeowYU6gILU3F3Lm9NxYBSo3j7MkjfsW6L5g6lwb6Izgg2PsgZ9TBSq+I6qZ36ZnHstlbeBuJA8RFzmJGnNjiNKi8Gyn3/rad8niKqBAdSXV8pBFrVXBF/AKezWDNisxvffzi/8AyVUA9EgoQMTiPqN4VE6yv1usEd4liY/ED3Knh0vgIGMUTx9/N4Pig9/JJdQxnSBubHmzf+KtU/pa3cxkspoYflDiVJXO6EAE58qAA6LdnIAuQQNdZpYflVjLe2XHUrrU9Kbm/jMfYtZUp12ZA7BcOADcljzAQA8Pa6V/3TredXT2dTooaa4ZHsgLuQrc5febPYk6E2QjsGkorwHKbE1SyAsCu9QpuO8EaeMJNGvU1bM3e2nlecpyjxQz08lwArWU6lRm0XMQCR1DcMxta5gFLaRXUXB7Lj4SbV6Gt+rvxsjTpuijfrr8bQyls2kBmLhgONxl7tPxnmVbGM5u1yes6/GaL40UkUE2VdbD3nO89/DuHfLGUyk409LYUKVFq3N85ltcDWw6zm93tmU/LxApFKiEP1iQbdoUKLzisNyzxqdOmiFBvVlLXHHNqL+EzqmJSq/OUxkV9cl75G95AeK3vbsImcmsYiG9ieUuINTOKzqSBezkX1PAECX/AN78V/nt/L+E5Rwb/P5SQpdZA85mmrdNW5TV39t2bjYk28hE/KusHzq5DFQGAJt0dNxJv195Ok5iwHvfr1jZV4n4xRbq8Ry2xDjJnt9mwPdpDqmLxmHpDECpmQEK9mJKE7g/C19Li4vp1Tj9n01zglgAoL3+rYb+2xsZXjKhbLUzlWNxTTObmnezApuymxBJNyyk7hLGPSX27HavKxsXh+ZdBmDK+ddxygj2SO3fe04+ri3RspBHw9Ifs3BO6s+UlFK5mHDNe1+oXFr7rwiphg5ygXtuzDdMZV6tY36MxMTrmIJmkldCLEXJI6LD8ZM4bXogaDgpI8L7pTh8FULWFN/L5XmZiJauVFamuaxQDsGq+Gka4tlCjha4msdhVN4Rk7SLA+J0Ei2xj7z0h284pPkpvLEJbFK/u/1fjHmy2zB/5in9yof+mKXv2Eae0jxc+f5y04rtv5wBAOr0hVOs24LEwzZVahO6Bu/Xf9eM01R/eUSrLr7Kn1liCxKbIptTRlxeHWpoTTdlIAIvlY5hZt2ltOuC4nY+JdlZaiErcApVa5uNbnMS1+0mQ2jgUZCXRRbL0lW7KMwuRoL6X0vOfwmzEq+x/M4Qns6SnXUaCaxyjwmsz3DefYWLOa9JHzWzHJTctY3GYspvrrrKG2RiVXKcMMoOawp01ANrX6KA7u2CLyYuMyuR3qDY8RmU6kdwlybCqL7OJdfshv8AUJvti4SbD11XL9HYC97g1VPVvRwbdkSYiqupRwQGAOavmswsQWaqT3R/omJXdja48XH/AFyQq4wbsW7fbXN/UWMVJcK02s4PS502FrM5t33ZGN/GDLj2uMzMw09rmW3d+HhQxmN41abfapL/APXJfTMT7yYV++m4+AWFD1cWhJKqiDSy8xSa1hYDMpQkWJ85FsRSIA5td1m/ZAC9soy5MRdRbv3wh8U/vYXDP9hyn9T/AClT4pBbPgW1+pVB7eCHqgHYTG4dC+QECqArWRzlt7Dksz2ylmGnAg+7N/DbTYMKlTO7KOgoBFEk73aouhW1rKxA7dLzkmxWDPtUsUh7FRgPNl+EgfoB/wDFqIf36f8ApZo6SlfKfaTYjEO5IIuALG4sBa9xvPbM9KhmqqYAaviXbsSm/wAWUSnG18EUIoDEZxazOEyHXW+t9190ktGwVQZrvbKvSPhu9bTPqYhq9TNbQnKiDqvbQdZMoxlWyZR72ngJocmFtWD/AOWj1P4kQlD98pAPxeNSi3MomdksKj5mF2GjLTG7Ip0vva3AWmWUCuxXRWs1uom9/kfGW4kszKxUAG2WwsbE2sTx/ES7E0RYMzhNOIY5rWuBlB6+NoUOW1kS0YOg35m7rL6m/wAJM4hPdpX+0zsf5co9IRXmiBubb5P6Uw3Ki/wqf6rmJ9o1SMvOPl6lYqvflFhfwkFO0SyAKyspbgwIJXfex4GF4yj+3KBTZQEFze/NjIGHVfLeYWIqZnN9Z6hyawqVXetVXNZUFtwLMyop8bE+EsJITZWFRULmnY1LMU5x8g3ZbISdbBdSSfhNBXUbkpjwZv6mIgJqj3d3Du4Rc6Zzl0H/AEthuIH2URT5hQZW+Nc6F3I7XY/OCXYxc2ZQ7VB1ekg1aSyDjH6PVApzGPLswigANQH1mk0QfvfehY5N4j3gV+26J/Wwli8nwPbrUV/9wv8A0K0zU+50oQgbyo72/OSDoN7r4An4CFLszDrvxF/sU2b+srLFpYUf5z+CU/8AVH/SoZ9Z0ZSvSYEWOlvjOPd6tB7XIsdCQLHiCLieiirRHs4cn7dRj/RljPigRlFCgB1FA/pUzXiKgufEPN6eOdWLK7BjvKmxPiIPjlaq+d3u1gLtcnTdradByi2M5bnaCqPrU0VUH2kVQFHaAJy5xTqbMNRvBBBHeOE6RlbExSxaLjc/lUt85p4bEPzfNNUK9ItmFmfjZc175dd15lDHDinkYZh6JdcwWw7WUX7rndNXEJVnrYiuvsVHbtKW9GBj09p4ni1/tIPkBINTy7wR3EH4EyBdes+oksp0A2zStqj37GW39MnhMVhqrhaiBUsTdn1JG5QwUBb9t72txnMmovXFzg6x5xaU3sSmGDNZGCg2HTDHv9kQVVoXtdstx0SLjeDqOvh+RMySw7Ig4ExGOUTczbtlljljERFT7jKyJmJVbLwHGIVQNyL36/jaBtXA4yh8V1TbmfE1Mz92k09juczqDbNSK+bobekxAZo7NfpqL2zXTxYEL/NlhXaNTXEVVZVC3ekoAtvLIm4bt0ly9wlOmKSpe+eqrd6CmDbsu0O5HB3f6S6ZKWHVmvr+1q2Kqq3G4XJtrY5Zy3LbaDF0ptq6IWfW1qlVucIIHUuQecSyyDVA4geUg1cncCfh6wEVG4ad34yQud8jQ6nSZt7Kvebn0g+0KTIRZ8wI3gWseqRRTJPTJFo7OgatrPSeT21KC4QKr/tWe7qzDNziginkQa82ty991yRvNp5+uDHWYfgVyG49ZUdogWWBxObTFtLVxZmdWrb7VRKzUEx/pTSaVHO4eO4ecUW0zUkdTA1qW9pvBPmxkztC3sqB2nU+ZkUXkP6EeZ/0t/rRRUjX5jrJ85IUV7IM2I7pBsRMrQ4ZRE1VRM1q5lD1jFDTfFgQHE7RtuMAqueswOpLGJaGPxbvpmNu+YlShNZ0MpanNxDMslqUiFImmaUgaUqM8sesyJvNI0pE0eyKGfaK0PNDsjfRoQBaK0O+jR/osKAih/0WP9FhAF5NH4Q36IJIYReqB1mH5fOaCJVU1HQBUJYBCR7LuALsR1XnI1yzuzuSzMSzE7ySbky5KAG4CTtAGWjJinL7RwhihUEkgstFOWJSuQALk7gNSe4QKAssRJ0GzuSWJq2OTm166nR/l9rzAm8nJrB4YZsVWztvyXyg9yLdz5zhl9Rx4zUTc+0dumPFlPdVHy4uhhmY5VBZupQSfIToMHyTxDi7JkH7xsfIazSr8r6VJcmEoKo+swCjvyrqfEic7tDbdev7bsVPujor90b/ABvJGXLl4iIj58ta4Y+Zv/QrG4WnR05xHccFGa3rYeMy3xBO8mUgSQE7YxMR3NuczE+IpLPHDRgI4E0iUUeKBpZpEtGvFaYbMzSl5eVMiUgCssqZIaaUgaUQgFklbJ2TQNKRNOaRnNSkOamgySDJKgHmoubhRWRKSoGyRskIKiMRAoyRc3LiY0CvLGtLcscJApjhJdaOqQKQkkKc2MBsDEVbFKbWPvHor5nf4XnTYHkIN9ep/DT/ANTD5TjyfUceHme28ePKfEOEFOa+A5NYmrYrTKr9ap0B66nwBnZHF4DB+wFZx9Qc4/i59nuuJi7R5cVWuKKKg+sek3roPIzj9/lz/jxqPeXT7eOP5T+heG5F0qa58TW0G8AhE7ix1PhaWPyhwWGBXDUw7brqMoPfUbpN6zisVi3qNmqOznrYk27urwlFo/xssu+XKZ+PEJ9yI/GK/tu7R5WYmrcB+bXqp6Hxbf5ETCYkm51J3k8Y4EkBPRjx44xWMUxlnll5lELJBZICSAm2UQskFjgSQgRCyQEe0eArRRRQNUU4+SSivMNo5IxWOzytqkBMsqaOzyl6ghDsZWxjM8rLSiTSpmEciNllRBjK2lxWNlltFGWLLLiIlW+4XiylOWPlmhg9lVapsiM3cL27zuHjOl2fyGc2NVwg6h0m/Aes5Z82GHmWscMsvEOKCX3azQwGxq1X2EYjrt0fvHT1no1HYWEw65nC6e9UIPoej6QPH8sqCdGmpcjdbor5nXyE88/VZZdceN/LrHDEd5TTLwHIU6Gs4H7qan7x0HkZsfRMDhNWyBh9bpv3hdbeAE5HaPKrE1bgPza9VPo+bb/WYbMTqdTH2eXk/kyqPaDfDH8Y/bt9ocuQNKFO/wC9U0Hgo/ETlto7br1vbqMR9UdFfujf4wC0Vp24/p+PDxDGXJlPqjFaStFad3NGPaPaOBAYCSAjgRwIDASQj2khAYCPeMTGgOTGLSJaVM8CzNFKM0UDdNSQNWKKYbVNWlbVY8UIrLSMUUoVooooDXkSY8UCI6pdTwxO/SKKJSHQbP5Ju1ixCjtOY+AGnrOlwfJzD0tWBc/vbvujTzvFFPlc/Pn33T3cfFh7LtobfoYcZTfQaKq/7ATl9octarXFNRTHX7Teug8oop3+m4cMo2mO3LmynGZpzWKxbubu7OetiT5dUpEUU90RER080zfk9o0UUoaK0UUocR7RooQ4EcCKKBKIRRQFETFFAYmQdoooFJa8jaKKA8UUUD//2Q=="
          />
        </div>
      </div>
    </div>
  );
};
