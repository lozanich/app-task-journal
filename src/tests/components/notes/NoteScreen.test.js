import "@testing-library/jest-dom";
import { mount } from "enzyme";
import configureStore from "redux-mock-store"; //ES6 modules
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { NoteScreen } from "../../../components/notes/NoteScreen";
import { setActiveNote } from "../../../actions/notes";

jest.mock("../../../actions/notes", () => {
  return { setActiveNote: jest.fn() };
});

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
  auth: {},
  ui: { loading: false, msgError: null },
  notes: {
    active: {
      id: 1234,
      title: "mi test title",
      body: "mi test body",
      date: 0,
    },
    notes: [],
  },
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <NoteScreen />
  </Provider>
);

describe("Pryebas en el <NoteScrren />", () => {
  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de diparar el active note", () => {
    const inputTitle = wrapper.find('input[name="title"]');
    inputTitle.simulate("change", {
      target: {
        name: "title",
        value: "Hola mi nuevo test",
      },
    });

    expect(setActiveNote).toHaveBeenCalled();
    expect(setActiveNote).toHaveBeenLastCalledWith(1234, {
      body: "mi test body",
      title: "Hola mi nuevo test",
      id: 1234,
      date: 0,
    });
  });
});
