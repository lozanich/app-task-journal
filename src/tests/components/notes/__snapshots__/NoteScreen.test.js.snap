// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pryebas en el <NoteScrren /> Debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      Wednesday, December 31st 1969, 6:00:00 pm
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Title"
      type="text"
      value="mi test title"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happend today?"
      value="mi test body"
    />
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Eliminar
    </button>
  </div>
</div>
`;
